source/common/http/http2/codec_impl.cc

```diff
@@ -441,7 +451,7 @@ void ConnectionImpl::StreamImpl::onPendingFlushTimer() {
 }
 
 void ConnectionImpl::StreamImpl::encodeData(Buffer::Instance& data, bool end_stream) {
-  ASSERT(!local_end_stream_);
+  ASSERT(!local_end_stream_ || (parent_.skip_encoding_empty_trailers_ && end_stream));
```

Comment:
Assuming that `local_end_stream_` means that we already sent `END_STREAM`, this seems wrong, since we cannot emit it again.

Reply:
It doesn't mean that, it's logical in that the stream owner thinks that have finished the stream. However, I think this is a bit fragile from an assert perspective. Can you add an `encodeDataHelper` or similar and pass a boolean as to whether we are calling this to handle the empty trailer case? That will all the assert to be more clear.

Reply:
Sorry I didn't get you for the first time. While I'm not 100% sure, I have updated to have a helper.